### PR TITLE
gpx export feature

### PR DESCRIPTION
**Pull request by MichaelsJP**
gpx export feature integrated

### CODE REVIEW

File: openrouteservice/src/main/java/heigit/ors/util/gpxUtil/Gpx.java [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+ * 		GPX documents contain a metadata header, followed by waypoints, routes, and tracks.  You can add your own elements
+ * 		to the extensions section of the GPX document.
+ *
+ *
+ * <p>Java class for gpxType complex type.
+ *
+ * <p>The following schema fragment specifies the expected content contained within this class.
+ * 
+ * <pre>
+ * &lt;complexType name="gpxType">
```
Comment (rabidllama) [inline openrouteservice/src/main/java/heigit/ors/util/gpxUtil/Gpx.java]: Should be < and not &lt

File: openrouteservice/src/main/java/heigit/ors/util/gpxUtil/LinkType.java [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+ *
+ * 	 A link to an external resource (Web page, digital photo, video clip, etc) with additional information.
+ *
+ *
+ * <p>Java class for linkType complex type.
+ *
+ * <p>The following schema fragment specifies the expected content contained within this class.
+ *
+ * <pre>
+ * &lt;complexType name="linkType">
```
Comment (rabidllama) [inline openrouteservice/src/main/java/heigit/ors/util/gpxUtil/LinkType.java]: < and not &lt;

File: openrouteservice/src/main/java/heigit/ors/util/gpxUtil/MetadataType.java [hunk 3]
```diff
@@ -0,0 +1,299 @@
...
+ * 		Information about the GPX file, author, and copyright restrictions goes in the metadata section.  Providing rich,
+ * 		meaningful information about your GPX files allows others to search for and use your GPS data.
+ *
+ *
+ * <p>Java class for metadataType complex type.
+ *
+ * <p>The following schema fragment specifies the expected content contained within this class.
+ * 
+ * <pre>
+ * &lt;complexType name="metadataType">
```
Comment (rabidllama) [inline openrouteservice/src/main/java/heigit/ors/util/gpxUtil/MetadataType.java]: < not &lt

File: openrouteservice/src/main/java/heigit/ors/util/gpxUtil/PersonType.java [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+ * 
+ * 	 A person or organization.
+ *
+ *
+ * <p>Java class for personType complex type.
+ *
+ * <p>The following schema fragment specifies the expected content contained within this class.
+ * 
+ * <pre>
+ * &lt;complexType name="personType">
```
Comment (rabidllama) [inline openrouteservice/src/main/java/heigit/ors/util/gpxUtil/PersonType.java]: < not &lt

File: openrouteservice/src/main/java/heigit/ors/util/gpxUtil/PtType.java [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+ * 
+ * 	 A geographic point with optional elevation and time.  Available for use by other schemas.
+ *
+ *
+ * <p>Java class for ptType complex type.
+ *
+ * <p>The following schema fragment specifies the expected content contained within this class.
+ * 
+ * <pre>
+ * &lt;complexType name="ptType">
```
Comment (rabidllama) [inline openrouteservice/src/main/java/heigit/ors/util/gpxUtil/PtType.java]: < not &lt

File: openrouteservice/src/main/java/heigit/ors/util/gpxUtil/PtType.java [hunk 6]
```diff
@@ -0,0 +1,146 @@
+
+package heigit.ors.util.gpxUtil;
+
+import javax.xml.bind.annotation.*;
```
Comment (rabidllama) [inline openrouteservice/src/main/java/heigit/ors/util/gpxUtil/PtType.java]: We should avoid importing all items from the package to keep things clean

File: openrouteservice/src/main/java/heigit/ors/util/gpxUtil/PtsegType.java [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+ * 
+ * 	 An ordered sequence of points.  (for polygons or polylines, e.g.)
+ *
+ *
+ * <p>Java class for ptsegType complex type.
+ *
+ * <p>The following schema fragment specifies the expected content contained within this class.
+ * 
+ * <pre>
+ * &lt;complexType name="ptsegType">
```
Comment (rabidllama) [inline openrouteservice/src/main/java/heigit/ors/util/gpxUtil/PtsegType.java]: < not &lt

File: openrouteservice/src/main/java/heigit/ors/util/gpxUtil/PtType.java (hunk 6 above)
Comment (TimMcCauley) [reply openrouteservice/src/main/java/heigit/ors/util/gpxUtil/PtType.java]: Yes please, let's think lean!

File: openrouteservice/src/main/java/heigit/ors/util/gpxUtil/LinkType.java [hunk 8]
```diff
@@ -0,0 +1,116 @@
+
+package heigit.ors.util.gpxUtil;
+
+import javax.xml.bind.annotation.*;
```
Comment (rabidllama) [inline openrouteservice/src/main/java/heigit/ors/util/gpxUtil/LinkType.java]: We should avoid importing all items from the package to keep things clean

File: openrouteservice/src/main/java/heigit/ors/util/gpxUtil/Gpx.java [hunk 9]
```diff
@@ -0,0 +1,251 @@
+package heigit.ors.util.gpxUtil;
+
+import javax.xml.bind.JAXBException;
+import javax.xml.bind.annotation.*;
```
Comment (rabidllama) [inline openrouteservice/src/main/java/heigit/ors/util/gpxUtil/Gpx.java]: We should avoid importing all items from the package to keep things clean

File: openrouteservice/src/main/java/heigit/ors/util/gpxUtil/CopyrightType.java [hunk 10]
```diff
@@ -0,0 +1,120 @@
+
+package heigit.ors.util.gpxUtil;
+
+import javax.xml.bind.annotation.*;
```
Comment (rabidllama) [inline openrouteservice/src/main/java/heigit/ors/util/gpxUtil/CopyrightType.java]: We should avoid importing all items from the package to keep things clean

File: openrouteservice/src/main/java/heigit/ors/services/routing/requestprocessors/json/JsonRoutingResponseWriter.java [hunk 11]
```diff
@@ -20,33 +20,21 @@
...
-import heigit.ors.routing.RouteResult;
-import heigit.ors.routing.RouteSegment;
-import heigit.ors.routing.RouteSegmentItem;
-import heigit.ors.routing.RouteStep;
-import heigit.ors.routing.RouteStepManeuver;
-import heigit.ors.routing.RouteSummary;
-import heigit.ors.routing.RoutingProfileType;
-import heigit.ors.routing.RoutingRequest;
-import heigit.ors.routing.WeightingMethod;
+import heigit.ors.routing.*;
```
Comment (rabidllama) [inline openrouteservice/src/main/java/heigit/ors/services/routing/requestprocessors/json/JsonRoutingResponseWriter.java]: We should avoid importing all items from the package to keep things clean

File: openrouteservice/src/main/java/heigit/ors/services/routing/requestprocessors/gpx/GpxRoutingResponseWriter.java [hunk 12]
```diff
@@ -0,0 +1,166 @@
...
+
+import com.graphhopper.util.shapes.BBox;
+import com.vividsolutions.jts.geom.LineString;
+import com.vividsolutions.jts.geom.Point;
+import heigit.ors.config.AppConfig;
+import heigit.ors.routing.*;
+import heigit.ors.services.routing.RoutingServiceSettings;
+import heigit.ors.util.AppInfo;
+import heigit.ors.util.GeomUtility;
+import heigit.ors.util.gpxUtil.*;
```
Comment (rabidllama) [inline openrouteservice/src/main/java/heigit/ors/services/routing/requestprocessors/gpx/GpxRoutingResponseWriter.java]: We should avoid importing all items from the package to keep things clean

File: openrouteservice/src/main/java/heigit/ors/services/routing/requestprocessors/gpx/GpxRoutingResponseWriter.java [hunk 13]
```diff
@@ -0,0 +1,166 @@
+package heigit.ors.services.routing.requestprocessors.gpx;
+
+
+import com.graphhopper.util.shapes.BBox;
+import com.vividsolutions.jts.geom.LineString;
+import com.vividsolutions.jts.geom.Point;
+import heigit.ors.config.AppConfig;
+import heigit.ors.routing.*;
```
Comment (rabidllama) [inline openrouteservice/src/main/java/heigit/ors/services/routing/requestprocessors/gpx/GpxRoutingResponseWriter.java]: We should avoid importing all items from the package to keep things clean

File: openrouteservice/src/main/java/heigit/ors/services/routing/requestprocessors/RoutingRequestParser.java [hunk 14]
```diff
@@ -18,35 +18,26 @@
...
 import heigit.ors.localization.LocalizationManager;
-import heigit.ors.routing.RouteExtraInfoFlag;
-import heigit.ors.routing.RouteInstructionsFormat;
-import heigit.ors.routing.RouteSearchParameters;
-import heigit.ors.routing.RoutingErrorCodes;
-import heigit.ors.routing.RoutingProfileType;
-import heigit.ors.routing.RoutingRequest;
-import heigit.ors.routing.WayPointBearing;
-import heigit.ors.routing.WeightingMethod;
+import heigit.ors.routing.*;
```
Comment (rabidllama) [inline openrouteservice/src/main/java/heigit/ors/services/routing/requestprocessors/RoutingRequestParser.java]: We should avoid importing all items from the package to keep things clean

**Review by rabidllama — CHANGES_REQUESTED** [review]
Mostly looks good, main thing I would say is that we should try to not use the .* for imports. It doesn't affect performance, but it can lead to issues down the line if there are clashes between packages.

Also, we should think about adding the unit testing to it.

**Comment by TimMcCauley** [comment]
@MichaelsJP let's add the unit tests, it's a good start.

**Review by rabidllama — APPROVED** [review]

**Pull request merged**